From 5c113340e3da56214d5c1c9451f86b5b95f34e9c Mon Sep 17 00:00:00 2001
From: Juan Gutierrez <juan.gutierrez@nxp.com>
Date: Sat, 3 Dec 2016 11:05:08 -0600
Subject: [PATCH 1333/5242] MXSCM-234 dts: imx: adjust the sd3 drive strength
 for 6sxscm evb

commit  597e498dcf763b7b486a68ca67ad95d04dd906b6 from
https://source.codeaurora.org/external/imx/linux-imx.git

For some SD Ultra (40MB/s) the drive strength/speed settings
on the iomux ctrl pads for SD3 is not enougn causing some
error by transfering data as below:

mmcblk2: error -84 transferring data, sector 2250553, nr 151,
cmd response 0x900, card status 0x0

Updating the DSE and Speed on the pad ctrl fixes the issue.

Signed-off-by: Juan Gutierrez <juan.gutierrez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts    |   14 ++++++++++++++
 .../imx6dqscm-1gb-evb-interleave-android-ldo.dts   |   14 ++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
index 03a6a90..04eacee 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-fix-ldo.dts
@@ -195,5 +195,19 @@
 				MX6QDL_PAD_EIM_D16__ECSPI1_SCLK 0x100b1
 			>;
 		};
+		pinctrl_usdhc3: usdhc3grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_CMD__SD3_CMD	0x170f9
+				MX6QDL_PAD_SD3_CLK__SD3_CLK	0x100f9
+				MX6QDL_PAD_SD3_DAT0__SD3_DATA0	0x170f9
+				MX6QDL_PAD_SD3_DAT1__SD3_DATA1	0x170f9
+				MX6QDL_PAD_SD3_DAT2__SD3_DATA2	0x170f9
+				MX6QDL_PAD_SD3_DAT3__SD3_DATA3	0x170f9
+				MX6QDL_PAD_SD3_DAT4__SD3_DATA4	0x170f9
+				MX6QDL_PAD_SD3_DAT5__SD3_DATA5	0x170f9
+				MX6QDL_PAD_SD3_DAT6__SD3_DATA6	0x170f9
+				MX6QDL_PAD_SD3_DAT7__SD3_DATA7	0x170f9
+			>;
+		};
 	};
 };
diff --git a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
index 72902e6..6b05421 100644
--- a/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
+++ b/arch/arm/boot/dts/imx6dqscm-1gb-evb-interleave-android-ldo.dts
@@ -186,6 +186,20 @@
 				MX6QDL_PAD_EIM_D16__ECSPI1_SCLK 0x100b1
 			>;
 		};
+		pinctrl_usdhc3: usdhc3grp {
+			fsl,pins = <
+				MX6QDL_PAD_SD3_CMD__SD3_CMD     0x170f9
+				MX6QDL_PAD_SD3_CLK__SD3_CLK     0x100f9
+				MX6QDL_PAD_SD3_DAT0__SD3_DATA0  0x170f9
+				MX6QDL_PAD_SD3_DAT1__SD3_DATA1  0x170f9
+				MX6QDL_PAD_SD3_DAT2__SD3_DATA2  0x170f9
+				MX6QDL_PAD_SD3_DAT3__SD3_DATA3  0x170f9
+				MX6QDL_PAD_SD3_DAT4__SD3_DATA4  0x170f9
+				MX6QDL_PAD_SD3_DAT5__SD3_DATA5  0x170f9
+				MX6QDL_PAD_SD3_DAT6__SD3_DATA6  0x170f9
+				MX6QDL_PAD_SD3_DAT7__SD3_DATA7  0x170f9
+			>;
+		};
 	};
 };
 
-- 
1.7.9.5

